src/orion/core/io/space_builder.py

```diff
@@ -337,10 +337,9 @@ def _build_from_args(self, cmd_args):
         recognise a ``--config`` option or by checking the first positional
         argument, if not found elsewhere. Only one is allowed.
 
-        .. note:: Currently positional arguments cannot define a parameter
-           dimension. And I don't think they will; it wouldn't be good for
-           your health down the line. Please be responsible and give a good
-           descriptive name to your inputs.
+        .. note:: Positional arguments cannot define a parameter
+           dimension, because no name, but an arbitrary (semantically) one, can
```

Comment:
I understand what you mean but it seems to me that it could be explained in more simple words. Just dropping "but an arbitrary (semantically) one" would make it more simpler. I don't think the note needs to justify why we do not assign random strings as hyper-parameter names when none are explicitly given.

Reply:
> We could have assigned `_5` as a name. Let's keep like this, please

Well that would be terrible. It would be very hard to understand what is `_5` when the user try to analyze the results latter on.

The documentation needs to be clear and concise. I'm sorry but this is not.

Reply:
`Positional arguments cannot define a parameter dimension, because no **meaningful** name can be assigned` ?

Reply:
:ok_hand: